scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskDataVariable.java

```diff
@@ -153,10 +156,27 @@ public void setGroup(String group) {
 
     @Column(name = "VARIABLE_ADVANCED")
     public Boolean getAdvanced() {
-        return advanced;
+        if (advanced == null) {
+            return false;
+        } else {
+            return advanced;
+        }
     }
 
     public void setAdvanced(Boolean advanced) {
         this.advanced = advanced;
     }
+
+    @Column(name = "VARIABLE_HIDDEN")
+    public Boolean getHidden() {
```

Comment:
I am a bit confused about boolean and Boolean, I don't know the difference and if it is harmful to use sometimes one and sometimes another.

Reply:
Boolean is an object, and thus can take a null value.
boolean is a primitive type and thus can only take true/false value.

In the database we want Boolean object as the column can have null values (workflows submitted with an earlier scheduler version).